symphony/lib/toolkit/class.database.php

```diff
@@ -62,6 +62,110 @@ public function getDatabaseErrorMessage()
 
 }
 
+class DatabaseStatement
+{
+    /**
+     * The database connection.
+     *
+     * @var Database
+     */
+    public $database;
+
+    /**
+     * The internal PDOStatement.
+     *
+     * @var PDOStatement
+     */
+    public $statement;
+
+    /**
+     * Wrap a PDOStatement object so that we can log its queries
+     * and handle its errors.
+     *
+     * @param Database      $database
+     * @param PDOStatement  $statement
+     */
+    public function __construct(Database $database, PDOStatement $statement)
+    {
+        $this->database = $database;
+        $this->statement = $statement;
+    }
+
+    /**
+     * Call a method on the internal PDOStatement object.
+     *
+     * @link    http://php.net/manual/en/class.pdostatement.php
+     *  For a list of available methods.
+     * @param   string $name
+     * @param   array  $arguments
+     * @return  mixed
+     */
+    public function __call($name, $arguments)
+    {
+        return $this->statement->{$name}(...$arguments);
```

Comment:
would it not be `call_user_func_array` then they can be passed through as they are?